include/circt-c/Dialect/FIRRTL.h

```diff
@@ -12,13 +12,119 @@
+// Dialect API.
+//===----------------------------------------------------------------------===//
+
 MLIR_DECLARE_CAPI_DIALECT_REGISTRATION(FIRRTL, firrtl);
 
+//===----------------------------------------------------------------------===//
+// Type API.
+//===----------------------------------------------------------------------===//
+
+/// Return 'true' if this is any FIRRTL type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsFIRRTLType(MlirType type);
+
+/// Return 'true' if this is a FIRRTL ground type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsGround(MlirType type);
+
+/// Return 'true' if this is a FIRRTL const type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsConst(MlirType type);
+
+/// Return 'true' if this is a FIRRTL Clock type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsClock(MlirType type);
+
+/// Return 'true' if this is a FIRRTL Reset type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsReset(MlirType type);
+
+/// Return 'true' if this is a FIRRTL AsyncReset type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsAsyncReset(MlirType type);
+
+/// Return 'true' if this is a FIRRTL SInt type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsSInt(MlirType type);
+
+/// Return 'true' if this is a FIRRTL UInt type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsUInt(MlirType type);
+
+/// Return 'true' if this is a FIRRTL Analog type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsAnalog(MlirType type);
+
+/// Return 'true' if this is a FIRRTL Bundle type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsBundle(MlirType type);
+
+/// Return 'true' if this is a FIRRTL Vector type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsFVector(MlirType type);
+
+/// Return 'true' if this is a FIRRTL Reference type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsRef(MlirType type);
+
+/// Return 'true' if this is a FIRRTL OpenBundle type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsOpenBundle(MlirType type);
+
+/// Return 'true' if this is a FIRRTL OpenVector type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsOpenVector(MlirType type);
+
+/// Return 'true' if this is a FIRRTL Enum type.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsFEnum(MlirType type);
+
+/// Return the bit-width of a type.
+MLIR_CAPI_EXPORTED int32_t firrtlTypeGetBitWidth(MlirType type,
+                                                 bool ignoreFlip);
+
+/// Return 'true' if the destination type is at least as wide as the source.
+MLIR_CAPI_EXPORTED bool firrtlTypeIsLarger(MlirType dst, MlirType src);
+
+/// Return 'true' if two types are equivalent.
+MLIR_CAPI_EXPORTED bool firrtlTypesAreEquivalent(MlirType dest, MlirType src,
+                                                 bool srcOuterTypeIsConst);
+
+//===----------------------------------------------------------------------===//
+// Aggregate Types
+//===----------------------------------------------------------------------===//
+
+/// Wrapped version of BundleType::BundleElement
+typedef struct {
+  MlirIdentifier name;
+  bool isFlip;
+  MlirType type;
+} FirrtlBundleField;
```

Comment:
See https://github.com/llvm/circt/pull/5472#issuecomment-1608794657

This struct has been added in that PR.